src/effects/backends/effectsbackendmanager.cpp

```diff
@@ -0,0 +1,115 @@
+        return;
+    }
+
+    m_effectsBackends.insert(pBackend->getType(), pBackend);
+
+    QList<QString> backendEffects = pBackend->getEffectIds();
+    for (const QString& effectId : backendEffects) {
+        m_manifests.append(pBackend->getManifest(effectId));
+    }
+
+    m_pNumEffectsAvailable->forceSet(m_manifests.size());
+
+    std::sort(m_manifests.begin(),
+            m_manifests.end(),
+            EffectManifest::alphabetize);
+}
+
+const QList<EffectManifestPointer> EffectsBackendManager::getManifestsForBackend(
+        EffectBackendType backendType) const {
+    auto pBackend = m_effectsBackends.value(backendType);
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return QList<EffectManifestPointer>();
+    }
+    auto list = pBackend->getManifests();
+    std::sort(list.begin(), list.end(), EffectManifest::alphabetize);
+    return list;
+}
+
+EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(
+        const QString& uid) const {
+    if (kEffectDebugOutput) {
+        //qDebug() << "EffectsBackendManager::getManifestFromUniqueId" << uid;
+    }
+    if (uid.isEmpty()) {
+        // Do not DEBUG_ASSERT, this may be a valid request for a nullptr to
+        // unload an effect.
+        return EffectManifestPointer();
+    }
+    int delimiterIndex = uid.lastIndexOf(" ");
+    EffectBackendType backendType =
+            EffectManifest::backendTypeFromString(uid.mid(delimiterIndex + 1));
+    VERIFY_OR_DEBUG_ASSERT(backendType != EffectBackendType::Unknown) {
+        // Mixxx 2.0 - 2.2 did not store the backend type in mixxx.cfg,
+        // so this code will be executed once when upgrading to Mixxx 2.3.
+        // This debug assertion is safe to ignore in that case. If it is
+        // triggered at any later time, there is a bug somewhere.
+        // Do not manipulate the string passed to this function, just pass
+        // it directly to BuiltInBackend.
+        return m_effectsBackends.value(EffectBackendType::BuiltIn)
+                ->getManifest(uid);
+    }
+    return m_effectsBackends.value(backendType)
+            ->getManifest(uid.mid(-1, delimiterIndex + 1));
+}
+
+EffectManifestPointer EffectsBackendManager::getManifest(
+        const QString& id, EffectBackendType backendType) const {
+    return m_effectsBackends.value(backendType)->getManifest(id);
+}
+
+const QString EffectsBackendManager::getDisplayNameForEffectPreset(
+        EffectPresetPointer pPreset) const {
+    QString displayName(QObject::tr("None"));
+    if (pPreset == nullptr || pPreset->isEmpty()) {
+        return displayName;
+    }
+
+    bool manifestFound = false;
+    for (const auto& pManifest : std::as_const(m_manifests)) {
+        if (pManifest->id() == pPreset->id() &&
+                pManifest->backendType() == pPreset->backendType()) {
+            displayName = pManifest->name();
+            manifestFound = true;
+            break;
+        }
+    }
+    DEBUG_ASSERT(manifestFound);
```

Comment:
if this fails, would that indicate failure to register an effect?  If so, this assert should print an error when it fails.

Reply:
DEBUG_ASSERT already does print an error on failure.